central/graphql/resolvers/node_components.go

```diff
@@ -159,15 +154,13 @@ func queryWithNodeIDRegexFilter(q string) string {
 		search.NewQueryBuilder().AddRegexes(search.NodeID, ".+").Query())
 }
 
-func (resolver *nodeComponentResolver) withNodeComponentScope(ctx context.Context) context.Context {
-	if env.PostgresDatastoreEnabled.BooleanSetting() {
-		return scoped.Context(ctx, scoped.Scope{
-			Level: v1.SearchCategory_NODE_COMPONENTS,
-			ID:    resolver.data.GetId(),
-		})
+func (resolver *nodeComponentResolver) nodeComponentScopeContext() context.Context {
+	if resolver.ctx == nil {
+		log.Errorf("attempted to scope context on nil")
+		return nil
 	}
-	return scoped.Context(ctx, scoped.Scope{
```

Comment:
We use nodeComponentResolver when postgres is disabled too. In that case, we convert image components to node components and wrap inside nodeComponentResolver. So we should keep both scopes.

https://github.com/stackrox/stackrox/blob/3a5c247b14aba154d27879ff5d01582be24324d6/central/graphql/resolvers/components_v2.go#L79